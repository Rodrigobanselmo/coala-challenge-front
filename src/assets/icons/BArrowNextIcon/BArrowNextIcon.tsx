import React, { FC } from "react";

import ArrowForwardIosIcon from "@mui/icons-material/ArrowForwardIos";
import { IIconMuiProps } from "@/core/interfaces/IIconMuiProps";

export const BArrowNextIcon: FC<IIconMuiProps> = ({ ...props }) => {
  return <ArrowForwardIosIcon {...props} />;
};

export default ArrowForwardIosIcon;
