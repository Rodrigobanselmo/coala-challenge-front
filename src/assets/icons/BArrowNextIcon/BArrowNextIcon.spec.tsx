import React from "react";
import { render, screen } from "@testing-library/react";
import BArrowNextIcon from "./BArrowNextIcon";

describe("BArrowNextIcon", () => {
  it("renders without errors", () => {
    render(<BArrowNextIcon />);
    const icon = screen.getByTestId("ArrowForwardIosIcon");

    expect(icon).toBeInTheDocument();
    expect(icon).toMatchSnapshot();
  });

  it("applies custom props", () => {
    render(<BArrowNextIcon fontSize="large" color="primary" />);

    const icon = screen.getByTestId("ArrowForwardIosIcon");
    expect(icon).toMatchSnapshot();
  });
});
