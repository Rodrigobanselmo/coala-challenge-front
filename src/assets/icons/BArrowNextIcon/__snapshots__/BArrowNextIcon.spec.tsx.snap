// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BArrowNextIcon applies custom props 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-bsqp9w-MuiSvgIcon-root"
  data-testid="ArrowForwardIosIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
  />
</svg>
`;

exports[`BArrowNextIcon renders without errors 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="ArrowForwardIosIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
  />
</svg>
`;
